File: src/main/java/twopointers/sumofthree/SumOfThree.java
package twopointers.sumofthree;

import java.util.Arrays;

public class SumOfThree {

    public boolean findSumOfThree(int[] nums, int target) {
        Arrays.sort(nums);
        int low, high, triples;
        for (int i = 0; i < nums.length - 2; i++) {
            low = i + 1;
            high = nums.length - 1;
            while (low < high) {
                triples = nums[i] + nums[low] + nums[high];
                if (triples == target) {
                    System.out.println(nums[i] + " + " + nums[low] + " + " + nums[high] + " = " + target);
                    return true;
                } else if (triples < target) {
                    low++;
                } else {
                    high--;
                }
            }
        }
        return false;
    }

}

File: src/main/kotlin/twopointers/sumofthree/SumOfThree.kt
package twopointers.sumofthree

fun isSumOfThree(nums: IntArray, target: Int): Boolean {
    nums.sort()
    nums.dropLast(1).forEachIndexed { index, _ ->
        run {
            var low = index + 1
            var high = nums.lastIndex
            while (low < high) {
                val triplet = nums[index] + nums[low] + nums[high]
                when (true) {
                    (triplet == target) -> {
                        println("${nums[index]} + ${nums[low]} + ${nums[high]} = $target")
                        return true
                    }
                    (triplet < target) -> low++
                    else -> high--
                }
            }
        }
    }
    return false
}
